### ISSUE TITLE
additional_data Causes Admin API to Fail

### ISSUE THREAD

**Issue by mk0y** [medusajs/medusa#9681]
### Describe the bug

I wanted to extend the product module and followed the docs here: https://docs.medusajs.com/v2/resources/commerce-modules/product/extend.

I added my custom props, like title_de (for titles in German language), and `curl` works with `additional_data`. But, when pass the body in the POST request without the `additional_data` the request will fail.

This also causes admin failing to publish a product saying something like "additional_data" is missing.

### System information

Medusa version (including plugins): rc
Node.js version: 20.17.0
Database: Postgres
Operating system: macos
Browser (if relevant): Not relevant, this is backend issue.

### Steps to reproduce the behavior

1. Extend the Product module as explained [here](https://docs.medusajs.com/v2/resources/commerce-modules/product/extend).
2. Make a POST request to create a product WITHOUT `additional_data` property.
3. See error `{"type":"invalid_data","message":"Invalid request: Field 'additional_data' is required"}`

### Expected behavior

CRUD operations on products should still function even without `additional_data` prop.

### Code snippets

This request fails:

```bash
curl -X POST 'localhost:9000/admin/products' \
-H 'Content-Type: application/json' \
-H 'Authorization: Bearer '$MY_JWT \
--data '{
    "title": "Shoes",
    "options": [
        {
            "title": "Default option",
            "values": ["Default option value"]
        }
    ]
}'
```

**Comment by olivermrbl** [comment]
@mk0y, thanks for the report. We'll look into this tomorrow and revert back. Your expectation of the behavior seems reasonable.

**Comment by mk0y** [comment]
Thanks @olivermrbl, i think it's the problem of middleware, my example:

```js
import { defineMiddlewares } from "@medusajs/medusa";
import { z } from "zod";

export default defineMiddlewares({
  routes: [
    {
      method: "POST",
      matcher: "/admin/products",
      additionalDataValidator: {
        title_de: z.string().nullable(),
        description_de: z.string().nullable(),
      },
    },
    {
      method: "POST",
      matcher: "/admin/products/:id",
      additionalDataValidator: {
        title_de: z.string().nullish(),
        description_de: z.string().nullish(),
      },
    },
  ],
});
```

Seems that `additionalDataValidator` isn't allowing anything anymore without `additional_data` property. Admin panel doesn't know about this so it fails.

**Comment by mk0y** [comment]
Unfortunately this issue was closed automatically without being resolved.

**Comment by olivermrbl** [comment]
@mk0y, this is solved in #9687 and will be part of the next release later today.
